### PR TITLE
chore: API and token registry renames

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

+ API cleanup
+ removing unnecessary functions (some functions were used only once - and that one time was in a test)
+ renaming

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/leverage/keeper/genesis.go [hunk 1]
```diff
@@ -79,17 +79,28 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
...
+	panicOnErr := func(err error) {
+		if err != nil {
+			panic(err)
+		}
+	}
+	tokens, err := k.GetAllRegisteredTokens(ctx)
+	panicOnErr(err)
+	reserves, err := k.GetAllReserves(ctx)
+	panicOnErr(err)
+	usuppy, err := k.GetAllUTokenSupply(ctx)
```
Comment (toteki) [inline x/leverage/keeper/genesis.go:91]:
```suggestion
	usupplies, err := k.GetAllUTokenSupply(ctx)
```

File: x/leverage/keeper/genesis.go [hunk 2]
```diff
@@ -79,17 +79,28 @@ func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
...
 		k.getAllAdjustedBorrows(ctx),
 		k.getAllCollateralSettings(ctx),
 		k.getAllCollateral(ctx),
-		k.GetAllReserves(ctx),
+		reserves,
 		k.GetLastInterestTime(ctx),
 		k.getAllBadDebts(ctx),
 		k.getAllInterestScalars(ctx),
-		k.GetAllUTokenSupply(ctx),
+		usuppy,
```
Comment (toteki) [inline x/leverage/keeper/genesis.go:103]:
```suggestion
		usupplies,
```

**Comment by toteki** [comment]
Propose `GetTokenSetting` -> `GetToken` rename. Thoughts?

File: x/leverage/keeper/keeper.go [hunk 3]
```diff
@@ -63,11 +63,22 @@ func (k Keeper) ModuleBalance(ctx sdk.Context, denom string) sdk.Int {
...
+func (k Keeper) validateLendAsset(ctx sdk.Context, loan sdk.Coin) error {
+	if !loan.IsValid() {
+		return types.ErrInvalidAsset.Wrap(loan.String())
+	}
+	token, err := k.GetTokenSettings(ctx, loan.Denom)
+	if err != nil {
+		return err
+	}
+	return token.AssertLendEnabled()
+}
```
Comment (toteki) [inline x/leverage/keeper/keeper.go:75]: This is good - I might look for other places to do something similar

File: x/leverage/gov_handler_test.go [hunk 4]
```diff
@@ -81,10 +81,10 @@ func TestUpdateRegistryProposalHandler(t *testing.T) {
 		tokens := k.GetAllRegisteredTokens(ctx)
 		require.Len(t, tokens, 3)
```
Comment (toteki) [inline x/leverage/gov_handler_test.go:82]:
```suggestion
		tokens, err := k.GetAllRegisteredTokens(ctx)
		require.NoError(t, err)
		require.Len(t, tokens, 3)
```

File: x/leverage/gov_handler_test.go [hunk 5]
```diff
@@ -81,10 +81,10 @@ func TestUpdateRegistryProposalHandler(t *testing.T) {
...
 
-		_, err := k.GetRegisteredToken(ctx, "uumee")
+		_, err := k.GetTokenSettings(ctx, "uumee")
```
Comment (toteki) [inline x/leverage/gov_handler_test.go:84]:
```suggestion
		_, err = k.GetTokenSettings(ctx, "uumee")
```

**Review by toteki — APPROVED** [review]
Reverted some changes (notably: `error` returns on store `Get` functions, where the only possible error is unmarshal, have changed back to `panic`)

Aside from that, should be good to merge.

**Comment by robert-zaremba** [comment]
thanks for updating the tests @toteki . But why did you revert back to panics? We should really avoid panics, even if it makes code more verbose. Having panics in API is a bad style, unfortunately we have too many of them in the Cosmos SDK.

**Comment by toteki** [comment]
Store+Getter+Unmarshal failures in particular should be extremely rare - hopefully never occur in production - that's why the code was initially written with panics in those places.

I'm following the sdk, where simple functions like (bank) GetBalance don't have an error return, even though they _technically_ could run into badly formatted data.

**Comment by robert-zaremba** [comment]
> Store+Getter+Unmarshal failures in particular should be extremely rare - hopefully never occur in production - that's why the code was initially written with panics in those places.

This is not a reason to use panics. Panic should be only used when there is no way to recover (ie, error won't make any sense).

> I'm following the sdk, where simple functions like (bank) GetBalance don't have an error return

It has been discussed various times, and we consider it a technical debt.

**Comment by robert-zaremba** [comment]
In general, there is no good argument for using panic in simple functions and errors in non simple functions.

**Pull request merged**